From a82e09c9d17093e5869ae7fa126cb04b046c404d Mon Sep 17 00:00:00 2001
From: "guoping.li" <guoping.li@amlogic.com>
Date: Fri, 10 Nov 2023 15:25:16 +0800
Subject: [PATCH 03/14] Multimedia: separate build output files for
 mediactrlsrc [1/2]

PD#SWPL-145539

Problem:
build output file *.o/*.so/bin is in the same file as the source code
it cause git error

Solution:
change the makefile and bb, set build output file to build/temp/work...

Verify:
local

Signed-off-by: guoping.li <guoping.li@amlogic.com>
Change-Id: Ic48e229b9f7bf5e891670934740fe68f9f087430
---
 recipes-multimedia/mediactrlsrc/mediactrlsrc.bb | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/recipes-multimedia/mediactrlsrc/mediactrlsrc.bb b/recipes-multimedia/mediactrlsrc/mediactrlsrc.bb
index f38cfa06..c63f5b01 100644
--- a/recipes-multimedia/mediactrlsrc/mediactrlsrc.bb
+++ b/recipes-multimedia/mediactrlsrc/mediactrlsrc.bb
@@ -8,7 +8,7 @@ PN = "${@bb.parse.BBHandler.vars_from_file(d.getVar('FILE'),d)[0] or 'mediactrls
 DEPENDS += "${@bb.utils.contains('DISTRO_FEATURES','mediactrlsrc-cam','isp','',d)}"
 DEPENDS += "${@bb.utils.contains('DISTRO_FEATURES','mediactrlsrc-hdmi','aml-tvserver','',d)}"
 
-EXTRA_OEMAKE = "EXTRA_CFLAGS+='-Wno-unused-result -Wno-unused-variable -Wno-unused-function'"
+EXTRA_OEMAKE = "OUT_DIR=${B} TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} EXTRA_CFLAGS+='-Wno-unused-result -Wno-unused-variable -Wno-unused-function -Os'"
 
 CAMERA_ENABLE = "${@bb.utils.contains('DISTRO_FEATURES','mediactrlsrc-cam','true','false',d)}"
 HDMIRX_ENABLE = "${@bb.utils.contains('DISTRO_FEATURES','mediactrlsrc-hdmi','true','false',d)}"
@@ -34,17 +34,18 @@ do_install() {
     install -d ${D}${bindir}
     install -d ${D}${libdir}
     install -d ${D}${includedir}
-    install -D -m 0644 ${S}/amlv4l2src/libamlv4l2src.so ${D}${libdir}/libamlv4l2src.so
-    install -D -m 0644 ${S}/amlv4l2src/amlsrc/libamlsrc.so ${D}${libdir}/libamlsrc.so
+
+    install -D -m 0644 ${B}/libamlv4l2src.so ${D}${libdir}/libamlv4l2src.so
+    install -D -m 0644 ${B}/libamlsrc.so ${D}${libdir}/libamlsrc.so
 
 	if [ ${CAMERA_ENABLE} = true ]; then
-		install -D -m 0755 ${S}/camctrl/camctrl ${D}${bindir}/camctrl
-		install -D -m 0644 ${S}/amlv4l2src/camsrc/libcamsrc.so ${D}${libdir}/libcamsrc.so
+		install -D -m 0755 ${B}/camctrl ${D}${bindir}/camctrl
+		install -D -m 0644 ${B}/libcamsrc.so ${D}${libdir}/libcamsrc.so
 	fi
 
 	if [ ${HDMIRX_ENABLE} = true ]; then
-		install -D -m 0644 ${S}/amlv4l2src/hdmisrc/libhdmisrc.so ${D}${libdir}/libhdmisrc.so
-		install -D -m 0755 ${S}/hdmictrl/hdmictrl ${D}${bindir}/hdmictrl
+		install -D -m 0644 ${B}/libhdmisrc.so ${D}${libdir}/libhdmisrc.so
+		install -D -m 0755 ${B}/hdmictrl ${D}${bindir}/hdmictrl
 	fi
 
     install -D -m 0644 ${S}/amlv4l2src/amlv4l2src.h ${D}${includedir}
-- 
2.25.1

